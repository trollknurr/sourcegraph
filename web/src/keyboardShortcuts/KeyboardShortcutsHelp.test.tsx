import React from 'react'
import renderer from 'react-test-renderer'
import { Modal } from 'reactstrap'
import { keybindingKeysDescription, KeyboardShortcutsHelp } from './KeyboardShortcutsHelp'

describe('KeyboardShortcutsHelp', () => {
    test('no user account', () => {
        const e = renderer.create(
            <KeyboardShortcutsHelp
                keyboardShortcuts={[
                    {
                        id: 'x',
                        title: 't',
                        keybindings: [{ held: ['Alt'], ordered: ['x'] }],
                    },
                ]}
                keyboardShortcutForShow={{
                    id: 'x',
                    title: 't',
                    keybindings: [{ held: ['Alt'], ordered: ['x'] }],
                }}
            />
        )
        // Modal is hidden by default and uses portal, so we can't easily test its contents. Grab
        // its inner .modal-body and snapshot that instead.
        expect(renderer.create(e.root.findByType(Modal).props.children[1]).toJSON()).toMatchSnapshot()
    })
})

describe('keybindingKeysDescription', () => {
    test('ordered only', () => expect(keybindingKeysDescription({ ordered: ['x'] })).toBe('x'))
    test('held and ordered', () =>
        expect(keybindingKeysDescription({ held: ['Control', 'Alt'], ordered: ['x'] })).toBe('Control+Alt+x'))
})
