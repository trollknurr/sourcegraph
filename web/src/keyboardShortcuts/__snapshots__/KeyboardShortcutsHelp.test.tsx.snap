// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyboardShortcutsHelp no user account 1`] = `
<div
  className="modal-body"
>
  <ul
    className="list-group list-group-flush"
  >
    <li
      className="list-group-item d-flex align-items-center justify-content-between"
    >
      t
      <span>
        <code
          className="badge badge-secondary mr-2"
        >
          Alt+x
        </code>
      </span>
    </li>
    <li
      className="list-group-item d-flex align-items-center justify-content-between"
    >
      Focus search bar
      <span>
        <code
          className="badge badge-secondary mr-2"
        >
          /
        </code>
      </span>
    </li>
    <li
      className="list-group-item d-flex align-items-center justify-content-between"
    >
      Expand URL to its canonical form (on file or tree page)
      <span>
        <code
          className="badge badge-secondary mr-2"
        >
          y
        </code>
      </span>
    </li>
  </ul>
</div>
`;
